src/components/molecules/Visualizer/useWidgetAlignSystem.ts

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState, useCallback } from "react";
+
+import type { WidgetLocationOptions } from "./Plugin/types";
+import type {
+  Widget,
+  WidgetAlignSystem,
+  WidgetSection,
+  WidgetZone,
+  WidgetArea,
+} from "./WidgetAlignSystem";
+
+export default ({ alignSystem }: { alignSystem: WidgetAlignSystem | undefined }) => {
+  const [overriddenAlignSystem, setOverrideAlignSystem] = useState<WidgetAlignSystem | undefined>(
+    alignSystem,
+  );
+
+  const moveWidget = useCallback((widgetId: string, options: WidgetLocationOptions) => {
+    if (
+      !widgetId ||
+      !["outer", "inner"].includes(options.zone) ||
+      !["left", "center", "right"].includes(options.section) ||
+      !["top", "middle", "bottom"].includes(options.area) ||
+      (options.section === "center" && options.area === "middle")
+    )
+      return;
+
+    let tarWidget: Widget | undefined;
```

Comment:
If this is short for targetWidget, just name it targetWidget since tar can mean different things

Reply:
I renamed it just as `widget` and also renamed `tarZone` `tarSection` `tarArea` to `targetZone` `targetSection` `targetArea`. Good to know that tar should not be used as shorten of target.

Reply:
For what it is worth, I understood what you meant :P